tests/test_lazy_login_and_share_times.py

```
@@ -0,0 +1,236 @@
+        import asyncio
+
+        cfg = {"base_url": "https://nas:5001", "username": "u", "password": "p"}
+        auth = MagicMock()
+        with patch("mcp_server.config.nas_configs", {"admin": cfg}), \
+             patch("mcp_server.config.get_synology_config", return_value=cfg), \
+             patch("mcp_server.SynologyAuth", return_value=auth):
+            # With an empty nas_name_map and allow_login=False, _get_base_url
+            # raises for the unmapped name rather than touching the NAS.
+            # Resolving it would trigger an on-demand login purely to tear the
+            # session down, which we must not do.
+            with pytest.raises(Exception):
```

Ruff (0.16.1)

[warning] Do not assert blind exception: `Exception`

(B017)